### DISCUSSION TITLE
AttributeError: numpy.ndarray object has no attribute properties

### DISCUSSION THREAD

**Discussion by xiaoxiongtao** [DeepTrackAI/DeepTrack2#298] [Resolved Issue]
I tried to run your code in the codelab, but I always got an error like "AttributeError: numpy.ndarray object has no attribute properties
". I tried other methods but none of them worked. I admire your work very much and hope you can help me solve this problem. xiaoxxtt@mails.cqjtu.edu.cn

![image](https://github.com/user-attachments/assets/1d25c7a7-65d7-4559-be47-f22d3042d3ee)

**Comment by shuyiloe** [comment]
In the newly released version of deeptrack, pipeline no longer returns image object but ndarray. If you need Image objects, you can enable them by using `pipeline.store_properties()`, restoring the behavior from previous releases.

**Comment by indirabarros** [comment]
How do you use it? It didn't work for me.

**Comment by harshithbachimanchi** [comment]
Hi! As @shuyiloe rightly pointed out, the [latest version](https://github.com/DeepTrackAI/DeepTrack2/releases/tag/2.0.0) of DeepTrack no longer returns `Image` objects (by default) which store the properties of feature's used to create the image. However, it can be enabled by using `.store_properties()` on the pipeline object before `.update()` step.

For this example, you could just do `dataset.store_properties()` before `dataset.update()` and that should solve the issue!
